server/task.go

```diff
@@ -297,7 +298,30 @@ func (s *Server) registerTaskRoutes(g *echo.Group) {
 			return echo.NewHTTPError(http.StatusInternalServerError, "Failed to update task status").SetInternal(err)
 		}
 		if task == nil {
-			return echo.NewHTTPError(http.StatusUnauthorized, fmt.Sprintf("Task ID not found: %d", taskID))
+			return echo.NewHTTPError(http.StatusNotFound, fmt.Sprintf("Task ID not found: %d", taskID))
+		}
+
+		issue, err := s.IssueService.FindIssue(ctx, &api.IssueFind{
+			PipelineID: &task.PipelineID,
+		})
+		if err != nil {
+			return echo.NewHTTPError(http.StatusInternalServerError, "Failed to find issue").SetInternal(err)
+		}
+		if issue == nil {
+			return echo.NewHTTPError(http.StatusNotFound, fmt.Sprintf("Issue not found by pipeline ID: %d", task.PipelineID))
+		}
+		if issue.AssigneeID == api.SystemBotID {
+			currentPrincipal, err := s.composePrincipalByID(ctx, currentPrincipalID)
+			if err != nil {
+				return echo.NewHTTPError(http.StatusInternalServerError, "Failed to find principal").SetInternal(err)
+			}
+			if currentPrincipal.Role != api.Owner && currentPrincipal.Role != api.DBA {
+				return echo.NewHTTPError(http.StatusUnauthorized, "Only allow Owner/DBA to update this task status")
+			}
+		} else {
+			if issue.AssigneeID != currentPrincipalID {
```

Comment:
With another thought, we can put these two requirements.
1) assignee should be owner or DBA; and 2) allow any owner, DBA, or issue creator.

This helps the issue to be taken over by another person if the current assignee isn't available at the moment.

Reply:
1. The issue assignee must be an Owner/DBA/Bot in the merged PR https://github.com/bytebase/bytebase/pull/849
2. Do you mean that we should allow any owner, DBA, or issue creator to approve issue? Or allow them to change the assignee?

This PR is attempting to prevent issue approve from unauthorised users.

Reply:
Gotit, I misread and thought it was for task update instead of task status update.

LGTM